x-pack/plugin/monitoring/src/main/java/org/elasticsearch/xpack/monitoring/exporter/http/HttpExporter.java

```diff
@@ -197,7 +197,20 @@ public void validate(final List<String> hosts, final Map<Setting<?>, Object> set
      */
     public static final Setting.AffixSetting<String> PROXY_BASE_PATH_SETTING =
             Setting.affixKeySetting("xpack.monitoring.exporters.","proxy.base_path",
-                    (key) -> Setting.simpleString(key, Property.Dynamic, Property.NodeScope));
+                    (key) -> Setting.simpleString(
+                        key,
+                        value -> {
+                            if (Strings.isNullOrEmpty(value) == false) {
```

Comment:
what happens if the value is null or empty ? I think the validation here should fail, else it may fail on cluster state application.

Reply:
Currently, the REST client builder does not set a proxy base path if the setting's value is null or empty. Do you think it should be changed so the setting itself refuses null or empty values?

Reply:
I just realized we actually need to support `null` as the way to un-set. If the underlying code already protects against empty (e.g. doesn't throw an exception) it should be fine as-is.